File: src/test/resources/testData/simpleLibrary/javaLibrary/ru/vldf/simplelibrary/Computer.java
package ru.vldf.simplelibrary;

import ru.vldf.simplelibrary.Memory;
import ru.vldf.simplelibrary.OS;

class Computer {
    Memory memory = null;
    boolean isOsLoad = false;

    void boot() {
        memory = new Memory();
    }

    void selectOs(String osName) {
        if (memory == null) {
            throw new IllegalStateException("Os wasn't loaded");
        }
        if (osName == "win") {
            memory.setOS(OS.WIN);
        } else if (osName == "linux") {
            memory.setOS(OS.LINUX);
        } else {
            throw new IllegalArgumentException("Invalid OS name");
        }
    }

    void loadOS() {
        isOsLoad = true;
    }

    void shutdown() {
        memory = null;
        isOsLoad = false;
    }
}

File: testLibrary/src/kotlin/ru/vldf/testlibrary/Computer.kt
package ru.vldf.testlibrary

class Computer {
    private var memory: Memory? = null
    private var isOsLoad = false

    fun boot() {
        memory = Memory()
    }

    fun selectOS(osName: String) {
        if (memory == null) {
            throw IllegalAccessException("Os wasn't loaded")
        }
        when (osName) {
            "win" -> memory!!.setOS(OS.WIN)
            "linux" -> memory!!.setOS(OS.LINUX)
            else -> throw IllegalArgumentException("Invalid OS name")
        }
    }

    fun loadOS() {
        isOsLoad = true
    }

    fun shutdown() {
        memory = null
        isOsLoad = false
    }
}